agent_api_rest/src/configurations/credential_configurations.rs

```diff
@@ -0,0 +1,191 @@
+use agent_issuance::{
+    server_config::{command::ServerConfigCommand, queries::ServerConfigView},
+    state::{IssuanceState, SERVER_CONFIG_ID},
+};
+use agent_shared::handlers::{command_handler, query_handler};
+use axum::{
+    extract::{Json, State},
+    http::StatusCode,
+    response::{IntoResponse, Response},
+};
+use oid4vci::{
+    credential_format_profiles::{CredentialFormats, WithParameters},
+    credential_issuer::credential_issuer_metadata::CredentialIssuerMetadata,
+};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use tracing::info;
+
+#[derive(Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CredentialConfigurationsEndpointRequest {
+    pub credential_configuration_id: String,
+    #[serde(flatten)]
+    pub credential_format_with_parameters: CredentialFormats<WithParameters>,
+    #[serde(default)]
+    pub display: Vec<serde_json::Value>,
+}
+
+#[axum_macros::debug_handler]
+pub(crate) async fn credential_configurations(
+    State(state): State<IssuanceState>,
+    Json(payload): Json<Value>,
+) -> Response {
```

Comment:
Volgens mij kan je `impl IntoResponse` hier gebruiken. dan kan je `map_err` in combinatie met ? gebruiken, misschien nog iets cleaner.

Reply:
Originally I implemented all the endpoints returning `impl IntoResponse` but in a review Tensor advised me to change it: https://github.com/impierce/ssi-agent/pull/26#discussion_r1506408055

I'm open to change it again eventually but I want to wait with that once we have aligned all our UniCore errors as well as OpenID4VC specific errors with the statuscodes we want to return.

Reply:
Ok I think into response is nice because it gives flexibility on your return statements and allows for use of the `?` operator on certain places. But for now its good :+1: